blocks/library/list/index.js

```diff
@@ -17,60 +17,45 @@ import { registerBlockType, source, createBlock } from '../../api';
 import Editable from '../../editable';
 import BlockControls from '../../block-controls';
 
-const { children, prop } = source;
+const { html, prop } = source;
 
 const fromBrDelimitedContent = ( content ) => {
 	if ( undefined === content ) {
 		// converting an empty block to a list block
 		return content;
 	}
-	const listItems = [];
-	listItems.push( createElement( 'li', [], [] ) );
-	content.forEach( function( element, elementIndex, elements ) {
-		// "split" the incoming content on 'br' elements
-		if ( 'br' === element.type && elementIndex < elements.length - 1 ) {
-			// if is br and there are more elements to come, push a new list item
-			listItems.push( createElement( 'li', [], [] ) );
-		} else {
-			listItems[ listItems.length - 1 ].props.children.push( element );
+
+	return content.split( /<br\s*?\/?>/ ).reduce( ( memo, item ) => {
+		if ( item ) {
+			memo.push( `<li>${ item }</li>` );
 		}
-	} );
-	return listItems;
+		return memo;
+	}, [] ).join();
 };
 
 const toBrDelimitedContent = ( values ) => {
 	if ( undefined === values ) {
 		// converting an empty list
 		return values;
 	}
+	const list = document.createElement( 'ul' );
+	list.innerHTML = values;
 	const content = [];
-	values.forEach( function( li, liIndex, listItems ) {
-		if ( typeof li === 'string' ) {
-			content.push( li );
-			return;
-		}
-
-		Children.toArray( li.props.children ).forEach( function( element, elementIndex, liChildren ) {
-			if ( 'ul' === element.type || 'ol' === element.type ) { // lists within lists
-				// we know we've just finished processing a list item, so break the text
-				content.push( createElement( 'br' ) );
-				// push each element from the child list's converted content
-				content.push.apply( content, toBrDelimitedContent( Children.toArray( element.props.children ) ) );
-				// add a break if there are more list items to come, because the recursive call won't
-				// have added it when it finished processing the child list because it thinks the content ended
-				if ( liIndex !== listItems.length - 1 ) {
-					content.push( createElement( 'br' ) );
-				}
+	const { TEXT_NODE } = window.Node;
```

Comment:
Could we keep browser dependencies at the top?

Reply:
sure